api-call.py

```diff
@@ -1,29 +1,226 @@
-import requests
+import datetime
 import json
 
+import pytz
+import requests
+import yaml
+
+
+def load_from_config():
+    """Loading important parameters from configuration file
+
+    :returns: The config object
+    """
+
+    with open('configuration.yaml', 'r') as file:
+        return yaml.safe_load(file)
+
+
+def log_error(error_message):
+    """Logging any error messages in the file and appending
+    with date time information in PST
+
+    :param error_message: The error message to log in file comes as a string
+    """
+
+    # Get the current datetime in PST
+    pst_timezone = pytz.timezone('America/Los_Angeles')
+    current_time = datetime.datetime.now(pst_timezone)
+    formatted_message = f"[{current_time}] {error_message}"
+
+    # Append the error message to the logfile
+    with open('logfile.txt', 'a') as file:
+        file.write(formatted_message + '\n')
+
+
+def get_access_token():
+    """Get access token from OAuth2 for connecting with OSU public APIs
+
+    :returns: The access token string
+    """
+
+    # Load configurations from the YAML file
+    config = load_from_config()
+    url = config['accessToken']['url']
+    enhancer = config['accessToken']['enhancer']
+    payload = config['accessToken']['payload']
+    headers = config['accessToken']['headers']
+    http_failed = config['httpResponses']['failed']
```

Comment:
What is `config['httpResponses']['failed']`?

Reply:
Th idea was to put strings in config/strings file that we can change the static stuff at one place, instead of finding them in the functions. It looks like this in yaml file:
httpResponses:
  failed: 'Request failed with status code: '

Reply:
We don't usually do that in the config file. If you would like a place to manage these error strings, instead of putting them into your config, you should have a constant file for them separately. For example: https://github.com/osu-mist/web-api-skeleton/blob/master/src/main/resources/edu/oregonstate/mist/api/resource.properties

Reply:
Got it.